Added T2 and iPad data

rickmark opened the pull request:
This tool already works out of box with iPad and T2 targets since the VDMs are identical

marcan commented on README.md:
```diff
@@ -15,17 +17,19 @@ This is based on portions of [ThunderboltPatcher](https://github.com/osy/Thunder
 
 Thanks to t8012.dev and mrarm for assistance with the VDM and Ace2 host interface commands.
 
+The documentation of the ACE and its USB-PD VPDs is here https://blog.t8012.dev/ace-part-1/
```
VDPs?

marcan commented on README.md:
```diff
@@ -15,17 +17,19 @@ This is based on portions of [ThunderboltPatcher](https://github.com/osy/Thunder
+Connect the two devices via their DFU ports. That's the furthest left port on M1 MacBooks, the closest left port on the T2, and the only port on the iPad.  See Apple's DFU recovery support articles to identify these ports on other models.
 
-You need to use a *USB 3.0 compatible* (SuperSpeed) Type C cable. USB 2.0-only cables, including most cables meant for charging, will not work, as they do not have the required pins. Thunderbolt cables work too.
+You need to use a *USB 3.0 compatible* (SuperSpeed) Type C cable. USB 2.0-only cables, including most cables meant for charging, will not work, as they do not have the required pins (USB CC1/CC2 where USB-PD are transmitted). Thunderbolt cables work too.
```
This is not correct; the missing pins are SBU1/SBU2, which we use for serial. USB 2.0 cables *do* work for USB-PD commands like reboot, just not for serial.

rickmark replied:
Honestly its missing both... but neither here nor there (it only ends up connecting D+/D-  SS1/2 and VCC / GND on a 2.0 cable)

marcan replied:
No, a 2.0 cable will have a CC line. A cable without a CC line connected is nonconformant and a lot of things would break with it, including direction and orientation and PD negotiation. I think you're confusing CC with "SS" (not sure what that is, either SBU or SSTX/SSRX?)

VinDuv committed:
README: Update for Monterey and new MBPs

- Indicate how to get the serial device on Monterey
- Document the location of the debug port on 14"/16" MBPs

Signed-off-by: Vincent Duvert <vincent@duvert.net>
Signed-off-by: Rick Mark <rickmark@outlook.com>

tihmstar commented:
Hi @marcan @rickmark,
I could successfully use this tool from my M1 mac to do: reboot, dfu and serial on my iPad8,1.
On T2 i could do: reboot and dfu, but serial doesn't work.
```
sudo macvdmtool serial
Mac type: J316cAP
Looking for HPM devices...
Found: IOService:/AppleARMPE/arm-io/AppleT600xIO/i2c0@9B040000/AppleS5L8940XI2CController/hpmBusManager@6B/AppleHPMBusController/hpm0/AppleHPMARMI2C
Connection: Source
Status: APP 
Unlocking... OK
Entering DBMa mode... Status: DBMa
Putting target into serial mode... VDM failed (reply: 0x05ac8092)
Exiting DBMa mode... OK
VDM failed
```
Any ideas?

On the same machine i can get serial from my iPad8,1.
Is it possible that serial VDM, is incorrect for T2?
How can the correct VMD be found?

rickmark committed:
Ignore output intermediates

rickmark commented:
> Hi @marcan @rickmark, I could successfully use this tool from my M1 mac to do: reboot, dfu and serial on my iPad8,1. On T2 i could do: reboot and dfu, but serial doesn't work.
> 
> ```
> sudo macvdmtool serial
> Mac type: J316cAP
> Looking for HPM devices...
> Found: IOService:/AppleARMPE/arm-io/AppleT600xIO/i2c0@9B040000/AppleS5L8940XI2CController/hpmBusManager@6B/AppleHPMBusController/hpm0/AppleHPMARMI2C
> Connection: Source
> Status: APP 
> Unlocking... OK
> Entering DBMa mode... Status: DBMa
> Putting target into serial mode... VDM failed (reply: 0x05ac8092)
> Exiting DBMa mode... OK
> VDM failed
> ```
> 
> Any ideas?
> 
> On the same machine i can get serial from my iPad8,1. Is it possible that serial VDM, is incorrect for T2? How can the correct VMD be found?

Serial only works if enabled by checkm8 / checkra1n - you'd have to reboot to DFU, then enable serial via the broken boot chain

graphine27 commented:
Hi @rickmark , how can serial be enabled on T2? I see no option in checkra1n and PongoOS only has a pinmux for ATV4K.
